### PR TITLE
settings: Simplify ClassValuesContextHandler

### PR DESCRIPTION
**Pull request by astaric**
Pull methods that can be used in other context handlers to
ContextHandler base class.

### CODE REVIEW

File: Orange/widgets/settings.py [hunk 1]
```diff
@@ -547,6 +547,11 @@ def close_context(self, widget):
     def settings_to_widget(self, widget):
         widget.retrieveSpecificSettings()
 
+        context = widget.current_context
+        if context is None:
+            return
+        self.provider.unpack(widget, context.values)
```
Comment (lanzagar) [inline Orange/widgets/settings.py]:
```py
if context is not None:
    self.provider.unpack(widget, context.values)
```

**Comment by kernc** [comment]
Would it make sense to add some more docstrings and usage instructions?

File: Orange/widgets/settings.py [hunk 2]
```diff
@@ -565,6 +570,18 @@ def packer(setting, instance):
...
 
+    def fast_save(self, widget, name, value):
+        super().fast_save(widget, name, value)
+
+        context = widget.current_context
+        if context is None:
+            return
+
+        if name in self.known_settings:
+            setting = self.known_settings(name)
```
Comment (lanzagar) [inline Orange/widgets/settings.py]: `self.known_settings[name]`

**Comment by astaric** [comment]
I have added docstrings for methods that did not have them.

As for the usage instructions, this is an abstract class with one abstract method. You need to create a subclass and provide the implementation of the match method and you are good to go.

**Comment by kernc** [comment]
Please excuse my seeming brain-rot (:flushed:), but I don't understand it, nor do I see code reflecting your statement. Which class is abstract? Which method is abstract? How do the things fit together? And why explain it to me over and over again and not also to countless Indians eager to have their contributions merged? If you dislike lengthy class or module docstrings, `doc/development/settings.rst` seems appropriate, though former can easily build the latter.

**Comment by janezd** [comment]
I don't really care about usage instructions here.

This PR has a different problem.

Open File -> Scatterplot. If you already have some saved settings, Scatterplot will fail with something like

    unable to set <PyQt4.QtGui.QComboBox object at 0x1154fff78> to value '('aquatic', 1)'.
    Possible values are hair, feathers, eggs, milk, airborne, aquatic, predator, toothed,
    backbone, breathes, venomous, fins, legs, tail, domestic, catsize, type.

If you change the values of the combo boxes or at least touch them, scatter plot starts working.

Now change the file. Every time you do so, scatter plot fail and the tuple accumulates. In the end, you can get something like

    File "/Users/janezdemsar/Dropbox/orange3/Orange/widgets/visualize/owscatterplotgraph.py",
    line 515, in update_data
        index_y = self.attribute_name_index[attr_y]
    KeyError: (((('aquatic', 1), -2), -2), -2)
    Pending nodes: set()
    Blocking nodes: set()
    unable to set <PyQt4.QtGui.QComboBox object at 0x1154fff78> to value '(((('aquatic', 1), -2), -2), -2)'.
    Possible values are hair, feathers, eggs, milk, airborne, aquatic, predator, toothed, backbone,
    breathes, venomous, fins, legs, tail, domestic, catsize, type.

I don't have time to dig into it, but apparently somewhere two values are packed in a tuple and not unpacked at retrieval.

**Comment by janezd** [comment]
I haven't really explored the code. Is it problematic in principle or is it just a bug and you're going to do it differently?

**Pull request closed without merging**